src/newgrf.cpp

```diff
@@ -8850,7 +8850,7 @@ void ResetPersistentNewGRFData()
  */
 static void BuildCargoTranslationMap()
 {
-	memset(_cur.grffile->cargo_map, 0xFF, sizeof(_cur.grffile->cargo_map));
+	std::fill(std::begin(_cur.grffile->cargo_map), std::end(_cur.grffile->cargo_map), uint8_t{UINT8_MAX});
```

Comment:
There is also `std::array::fill`

Reply:
There is. Should we keep it generic or keep it short? I found that so far all our uses of this particular `std::fill` pattern are used on C-arrays, and `std::array::fill` isn't used at all. I do however have some other WIP patches that use this pattern that could use the shorter form if we want.